spec/abci++/abci++_methods_002_draft.md

```diff
@@ -318,8 +318,11 @@ title: Methods
             * If the Application wants to add a new transaction, then the Application should include it in `tx_records` and _mark_ it as `ADD`. In this case, Tendermint will add it to the mempool.
         * The Application should be aware that removing and adding transactions may compromise _traceability_.
           > Consider the following example: the Application transforms a client-submitted transaction `t1` into a second transaction `t2`, i.e., the Application asks Tendermint to remove `t1` and add `t2` to the mempool. If a client wants to eventually check what happened to `t1`, it will discover that `t_1` is not in the mempool or in a committed block, getting the wrong idea that `t_1` did not make it into a block. Note that `t_2` _will be_ in a committed block, but unless the Application tracks this information, no component will be aware of it. Thus, if the Application wants traceability, it is its responsability to support it. For instance, the Application could attach to a transformed transaction a list with the hashes of the transactions it derives from. 
-        * If the Application modifies the set of transactions, the modified transactions MUST NOT exceed the configured maximum size `RequestPrepareProposal.max_tx_bytes`.
     * If the Application does not modify the preliminary set of transactions `txs`, then it sets `ResponsePrepareProposal.modified_tx_status` to `UNMODIFIED`. In this case, Tendermint will ignore the contents of `ResponsePrepareProposal.tx_records`.
+    * In order to optimize the block size in busy blockchains, Tendermint might include a list of transactions in `RequestPrepareProposal.txs` whose size
+      in bytes exceeds `RequestPrepareProposal.max_tx_bytes`. In this case:
+        * The Application MUST still abide by the `RequestPrepareProposal.max_tx_bytes` limit in the list of transaction records it returns (excluding those marked as `REMOVE`).
+        * It is _invalid_ to return `ResponsePrepareProposal.modified_tx_status` set to `UNMODIFIED`: Tendermint will panic.
```

Comment:
This sentence is a bit unclear to me. Does it mean that if Tendermint abuses `RequestPrepareProposal.txs` then the Application cannot set `ResponsePrepareProposal.modified_tx_status` to `UNMODIFIED`? 

I think it would be simpler to mention this more generally and not in the context of busy blockchains (also is this a well-known term?). Simply saying that if the Application modifies the set of txs, then the Application cannot set `ResponsePrepareProposal.modified_tx_status` to `UNMODIFIED`, otherwise Tendermint will panic. This covers all cases:
- the case when Tendermint provides a set of txs within the limit but the Application modifies it for some application -specific reasons.
- the busy blockchains case.

Reply:
The point is that, if Tendermint sends txs taking more than `max_tx_bytes`, the Application cannot return `UNMODIFIED` even if it doesn't modify them.
But I see the wording is not clear enough. Let me see what I can do...

Reply:
In my understanding, shaving transactions off `RequestPrepareProposal.txs` is modifying the set of transactions. But this is the way I understand it...

Reply:
That's correct.
I've reworked the wording of this part. Let me know if you find it clearer now.